### PR TITLE
support for python ecosystem through requirements.txt

### PR DESCRIPTION
**Pull request by invincibleJai**
Provides support for python ecosystem through requirements.txt, like we show recommendations currently for pom.xml.

### CODE REVIEW

File: src/collector.ts [hunk 1]
```diff
@@ -71,6 +71,73 @@ class DependencyCollector implements IDependencyCollector {
...
+        this.objSam = objSam;
+        this.parser = this.createPyParser()
+    }
+
+    objSam: any;
+    parser: Xml2Object;
+    dependencies: Array<IDependency> = [];
+    isDependency: boolean = false;
+    versionStartLine: number = 0;
+    versionStartColumn: number = 0;
```
Comment (msrb) [inline src/collector.ts]: I think these two fields `versionStartLine` and `versionStartColumn` are not used anywhere?

File: src/collector.ts [hunk 2]
```diff
@@ -71,6 +71,73 @@ class DependencyCollector implements IDependencyCollector {
...
 }
 
+class NaivePyParser {
+    constructor(objSam: any) {
+        this.objSam = objSam;
+        this.parser = this.createPyParser()
+    }
+
+    objSam: any;
+    parser: Xml2Object;
```
Comment (msrb) [inline src/collector.ts]: The type here is correct (?)

File: src/collector.ts [hunk 3]
```diff
@@ -71,6 +71,73 @@ class DependencyCollector implements IDependencyCollector {
...
+                deps.push(dep);
+        });
+    }
+
+     parse(): Array<IDependency> {
+        return this.dependencies;
+    }
+}
+
+let toObject = (arr:any) => {
```
Comment (msrb) [inline src/collector.ts]: I think `requirement.txt` files can be more complex and thus this code probably won't cover everything, but it's a start.

File: src/collector.ts (hunk 1 above)
Comment (invincibleJai) [inline src/collector.ts]: yes it's unused, will remove it

File: src/collector.ts (hunk 3 above)
Comment (invincibleJai) [inline src/collector.ts]: true , it'll be incremental approach.

**Comment by msrb** [comment]
LGTM, thanks :wink:

**Pull request merged**